src/v/coproc/script_context.cc

```diff
@@ -65,6 +69,32 @@ script_context::script_context(
       "valid subscription list");
 }
 
+static void exception_filterer(const std::exception_ptr& eptr) {
+    try {
+        /// If an exception is unhandled by this method, it will percolate to
+        /// the caller of start(). ATM by design only one exception should do
+        /// this, the \ref script_failed_exception.
+        std::rethrow_exception(eptr);
+    } catch (const std::logic_error& e) {
+        vlog(
```

Comment:
I don't see how this makes sense to me. we *should* expect - not abnormal- that wasm would crash constantly. why is it an exceptional case. 

why not use std::errc and error_category and enumerate the error paths.